local.go

```diff
@@ -276,32 +277,41 @@ func checkConvergence(loc *Location, itertype IterationType, stats *Stats, setti
 	return NotTerminated
 }
 
-// evaluate evaluates the function and stores the answer in place
+// evaluate evaluates the function given by funcs at xNext, stores the answer
+// into location and updates stats. If location.X is not equal to xNext and
+// evalType == NoEvaluation, location is filled with NaN values. If the
+// function does not support the requested evalType, location and stats are not
```

Comment:
Well, it does not panic yet, only when the evaluation type is unknown. If the evaluation type does not match the function, it returns an error, but I will change it to a panic.